### DISCUSSION TITLE
How to get a sum of element average values with element IDs

### DISCUSSION THREAD

**Discussion by NorthMagic** [idaholab/moose#26000] [Q&A General] [answered]
Dear MOOSE collaborators,

I try to solve equations with user-defined moose kernels, and the solved physical variable is element dependent. But I found that moose variables are basically indexed by _qp, and I don't know how to get an element average value by using an element ID. Now, I want to get a sum of element average values with a lot of known element IDs, could you give me some suggestions?

**Comment by lindsayad** [comment]
> I don't know how to get a elmental value

What do you mean? Do you want to get the average value on an element? Generally speaking, except for a small subset of orders (e.g. CONSTANT), the finite element approximate solution varies within an element

> by using a mesh ID.

Do you mean an element ID?

**Reply by NorthMagic** [reply]
I want to get the average value on a lot of elements. I know how to get the average value on an element, but how to get the average value of a lot of elements makes me confused. 
I'm sorry for expressing it equivocally. Mesh ID means an element ID.

**Comment by GiudGiud** [comment, marked as answer]
You will want to define a constant monomial (one value per element) auxiliary variable then use a ProjectionAux to project from the 'physical variable' to this constant monomial auxiliary variable
